.gitignore

```diff
@@ -1,6 +1,25 @@
 # Compiled class file
 *.class
 
+# Gradle
+/.gradle/
+/build/
+
+# Maven
+/target/
+
+# Generated by MacOS
+.DS_Store
+
+# Generated by Windows
+Thumbs.db
+
+# JetBrains IDE
+/.idea/*
+!/.idea/copyright
+!/.idea/codeStyles
```

Comment:
Maybe better to use editorconfig instead of /.idea/codeStyles? I don't think that is good idea to keep something from .idea in repo. We don't need to keep IDE related files.

Reply:
Exported code styles to .editorconfig.